src/Microsoft.OData.Core/TypeNameOracle.cs

```diff
@@ -86,12 +89,17 @@ internal static IEdmStructuredType ResolveAndValidateTypeFromTypeName(IEdmModel
             }
 
             // TODO: Clean up handling of expected types/sets during writing
-            var typeFromResource = (IEdmStructuredType)ResolveAndValidateTypeName(model, typeName, EdmTypeKind.None, /* expectStructuredType */ true, writerValidator);
+            var typeFromResource = ResolveAndValidateTypeName(model, typeName, EdmTypeKind.None, /* expectStructuredType */ true, writerValidator);

```

Comment:
Yes We Can!

---
In reply to: [127301025](https://github.com/OData/odata.net/pull/894#discussion_r127301025) [](ancestors = 127301025)